### PR TITLE
Remove dependency on annotation-tools

### PR DESCRIPTION
**Pull request by wmdietl**

### CODE REVIEW

**Review by cpovirk — APPROVED** [review]

File: checker/bin-devel/build.sh [hunk 1]
```diff
@@ -33,16 +33,6 @@ fi
...
-AT=$(dirname "${AFU}")
-
-## Build annotation-tools (Annotation File Utilities)
-"$PLUME_SCRIPTS/git-clone-related" typetools annotation-tools "${AT}" -q
-if [ ! -d ../annotation-tools ] ; then
-  ln -s "${AT}" ../annotation-tools
-fi
-
 echo "Checking out the annotation-tools commit at which jspecify last merged from upstream."
 git -C "${AT}" checkout -q 7174291c828e88382758e0d5117f99418970f24f
```
Comment (netdpb) [inline checker/bin-devel/build.sh:37]: Don't forget to delete these lines too.

File: checker/build.gradle [hunk 2]
```diff
@@ -4,9 +4,16 @@ plugins {
...
+        java {
+            // NO-AFU
+            exclude '**/org/checkerframework/checker/resourceleak/MustCallInference.java'
+        }
         resources {
             // Stub files, message.properties, etc.
             srcDirs += ['src/main/java']
+
+            // NO-AFU
+            exclude '**/org/checkerframework/checker/resourceleak/MustCallInference.java'
```
Comment (netdpb) [inline checker/build.gradle:16]: I see we've mixed resources in with the Java code. Should this source set just exclude `**/*.java`? Or do we expect the source files to be included in the compiled jar?

File: framework/build.gradle [hunk 3]
```diff
@@ -7,9 +7,21 @@ ext {
...
+            exclude '**/org/checkerframework/framework/stub/ToIndexFileConverter.java'
+        }
+
         resources {
             // Stub files, message.properties, etc.
             srcDirs += ['src/main/java', "${buildDir}/generated/resources"]
+
+            // NO-AFU
+            exclude '**/org/checkerframework/common/wholeprograminference/**'
+            exclude '**/org/checkerframework/framework/stub/AddAnnotatedFor.java'
```
Comment (netdpb) [inline framework/build.gradle:23]: Same comment as before about excluding Java files from the resources.

**Review by netdpb — COMMENTED** [review]
I'm not knowedgable enough about the implementation to review the Java changes.

File: framework/build.gradle (hunk 3 above)
Comment (wmdietl) [reply framework/build.gradle:23]: I removed these lines in `checker`, but when I do the same here, the javadoc build fails. I'll investigate separately.

**Pull request merged**